base/compiler/abstractinterpretation.jl

```diff
@@ -2321,9 +2334,7 @@ function typeinf_local(interp::AbstractInterpreter, frame::InferenceState)
                 if isa(lhs, SlotNumber)
                     changes = StateUpdate(lhs, VarState(t, false), changes, false)
                 elseif isa(lhs, GlobalRef)
-                    tristate_merge!(frame, Effects(EFFECTS_TOTAL,
-                        effect_free=TRISTATE_UNKNOWN,
-                        nothrow=TRISTATE_UNKNOWN))
+                    abstract_eval_global_assignment(interp, frame, lhs, t)
```

Comment:
`abstract_eval_xxx` usually get return type information back, which is already computed here, so maybe we want to name this as `handle_global_assignment!` (after `handle_control_backedge!`) or something?

Reply:
Sure, that seems reasonable.